enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,426 @@
+
+#### Representation
+
+A new opm sub-command, `represent` will be introduced that will output a yaml/json representation of an index, i.e, `opm represent --output=json docker.io/my-namespace/my-index` will output the following json representation of the index:
+
+```output.json
+{
+    "graph": {
+        "nodes": [
+            {
+                "bundlePath": "quay.io/etcd",
+                "provides": [
+                    {
+                        "type": "olm.package",
+                        "version": "1.0.0",
+                        "packageName": "etcd"
+                    },
+                    {
+                        "type": "olm.gvk",
+                        "version": "v1beta2",
+                        "group": "etcd.database.coreos.com",
+                        "kind": "EtcdCluster"
+                    },
+                    {
+                        "type": "label",
+                        "value": "alpha"
+                    }
+                ],
+                "constraints": [
+                    [
+                        {
+                            "type": "label",
+                            "value": "LTS"
+                        }
+                    ]
+                ]
+            },
+        ]
+    }
+}
+```
+
+When the `represent` command is summoned for an index, `opm` will make a temp copy of `definition.CUE`, extract the bundles from an index, and for each bundle, append the following blob to the copy of `definition.CUE`: 
+
+```
+// Add a node
+_nodes: {
+    <package-name>: {
+		bundlePath: <bundle-path>
+		provides: [
+            #PackageProvides & {
+                packageName: <package-name>,
+                version: <version>,
+            },
+            #GVKProvides & {
+				group:   <group>
+				kind:    <kind>
+				version: <version>
+			},
+            ...
+        ],
+        constraints: [
+			[#GVKConstraint & json.Unmarshal(#"{"group":<group>,"version":<version>, "kind":<kind>}"#), ...],
+      [#LabelConstraint & json.Unmarshal(#"{"type":<type>,"value":<value>}"#), ...]
+		]
+	},
+}
+```
+For example, for the example index above that has one bundle in it, the following blob will be appended to the copy of `definition.CUE` 
+
+```
+_nodes: {
+    "etcd-1.0.0": {
+		bundlePath: "quay.io/etcd"
+		provides:   [
+            #PackageProvides & {
+                packageName: "etcd",
+                version: "1.0.0",
+            },
+            #GVKProvides & {
+				group:   "etcd.database.coreos.com"
+				kind:    "EtcdCluster"
+				version: "v1beta2"
+			},
+            #LabelConstraint & {
+				value: "alpha"
+			},
+        ],
+		constraints: [
+            [
+			    #LabelConstraint & {
+				    value: "LTS"
+			    },
+            ],
+		]
+	},
+}
+```
+
+The final result of the above operation will then be [exported using CUE](https://github.com/cuelang/cue/blob/v0.2.2/cmd/cue/cmd/export.go), which will output the json representation of the index.
+
+
+#### Validation
```

Comment:
How we will run the validation against cue files? Are we going to import the cue libraries into `opm` or shell out to the cue binary directly? It's worth mentioning, I discuss this in #43. I also listed some caveats to using cue that could be helpful here as well.

Reply:
@exdx talked about importing the libraries we want to use from CUE [here](https://github.com/operator-framework/enhancements/pull/37/files#diff-6a0f94b7c8c55fc50c16dacdd2d64416R306)

> I also listed some caveats to using cue that could be helpful here as well.

Added a link to the section you discuss the caveats in right below that :arrow_up: